tests/FSharp.Test.Utilities/ReflectionHelper.fs

```diff
@@ -0,0 +1,63 @@
+module FSharp.Test.ReflectionHelper
+
+open System
+open System.Reflection
+
+/// Gets the given type from the assembly (otherwise throws)
+let getType typeName (asm: Assembly) =
+    match asm.GetType(typeName, false) with
+    | null ->
+        let allTypes =
+            asm.GetTypes()
+            |> Array.map (fun ty -> ty.Name)
+            |> Array.reduce (fun x y -> $"%s{x}\r%s{y}")
+
+        failwith $"Error: Assembly did not contain type %s{typeName}.\nAll types in asm:\n%s{allTypes}"
+    | ty -> ty
+
+/// Gets all anonymous types from the assembly
+let getAnonymousTypes (asm: Assembly) =
+    [ for ty in asm.GetTypes() do
+          if ty.FullName.StartsWith "<>f__AnonymousType" then ty ]
+
+/// Gets the first anonymous type from the assembly
+let getFirstAnonymousType asm =
+    match getAnonymousTypes asm with
+    | ty :: _ -> ty
+    | [] -> failwith "Error: No anonymous types found in the assembly"
+
+/// Gets a type's method
+let getMethod methodName (ty: Type) =
+    match ty.GetMethod(methodName) with
+    | null -> failwith $"Error: Type did not contain member %s{methodName}"
+    | methodInfo -> methodInfo
+
+/// Assert that function f returns Ok for given input
+let should f x y =
+    match f x y with
+    | Ok _ -> ()
+    | Error message -> failwith $"%s{message} but it should"
```

Comment:
should... should what? :)

Reply:
It should *do something*, *have something*, `<verb> <something>`, So expected error message is that `"<it> doesn't <verb> whatever"` - but it should! 😃 

In this case `"<Thing> doesn't have attribute <attribute name> but it should"` 

I'm hoping this will be reusable. TBH I'm more concerned that it's fixed to two parameters than with the error message.

Reply:
Well if anyone else starts using that, they'll probably adjust the framework anyway :)